server/src/main/java/org/elasticsearch/action/StatsLevel.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action;
+
+import org.elasticsearch.common.Strings;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public enum StatsLevel {
+    CLUSTER,
+    NODE,
+    INDICES,
+    SHARDS;
+
+    public static ActionRequestValidationException genIllegalClusterLevelException(String level) {
```

Comment:
I would prefer that we had two different enums, one without the `NODE` and the other without the `CLUSTER`. That way the type system tells us that we don't need to worry about handling an invalid value, which avoids much of the extra validation steps in this PR.